lib/modules/manager/nix/artifacts.ts

```diff
@@ -20,23 +21,28 @@ export async function updateArtifacts({
     return null;
   }
 
-  let cmd: string;
+  let cmd = `nix \
+    --extra-experimental-features nix-command \
+    --extra-experimental-features flakes `;
+
+  const { token } = hostRules.find({
+    hostType: 'github',
+    url: 'https://api.github.com/',
+  });
+
+  if (token) {
+    cmd += `--extra-access-tokens github.com=${token} `;
```

Comment:
i would prefer to pass it via `NIX_CONFIG` env 🤔

Reply:
That's okay for the GitHub app, but harder to do if someone is self-hosting. This way is potentially worse for logging, if you log shell commands, right?

Reply:
Why is env harder for self-hosting?

Reply:
Not env in general, but if someone has configured `NIX_CONFIG` we might overwrite it, or make it invalid.